src/python/doc/ripsnet.rst

```diff
@@ -0,0 +1,98 @@
+:orphan:
+
+.. To get rid of WARNING: document isn't included in any toctree
+
+RipsNet user manual
+=========================
+Definition
+----------
+
+.. include:: ripsnet.inc
+
+:class:`~gudhi.ripsnet` constructs a Tensorflow model for fast and robust estimation of persistent homology of
+point clouds.
+RipsNet is based on a Deep Sets architecture :cite:`DeepSets17`, for details see the paper RipsNet :cite:`RipsNet_arXiv`.
+
+Example
+-------------------
+
+This example instantiates a RipsNet model which can then be trained as any tensorflow model.
+
+.. testcode::
+
+    from gudhi.tensorflow import *
+    import tensorflow as tf
+    from tensorflow.keras import regularizers, layers
+
+    ragged_layers_size = [20, 10]
+    dense_layers_size = [10, 20]
+    output_units = 25
+    activation_fct = 'gelu'
+    output_activation = 'sigmoid'
+    dropout = 0
+    kernel_regularization = 0
+
+    ragged_layers = []
+    dense_layers = []
+
+    for n_units in ragged_layers_size:
+        ragged_layers.append(DenseRagged(units=n_units, use_bias=True, activation=activation_fct))
+
+    for n_units in dense_layers_size:
+        dense_layers.append(layers.Dense(n_units, activation=activation_fct,
+                                         kernel_regularizer=regularizers.l2(kernel_regularization)))
+        dense_layers.append(layers.Dropout(dropout))
+
+    dense_layers.append(layers.Dense(output_units, activation=output_activation))
+
+    phi_1 = TFBlock(ragged_layers)
+    perm_op = 'mean'  # can also be 'sum'.
+    phi_2 = TFBlock(dense_layers)
+    input_dim = 2
+
+    RN = RipsNet(phi_1, phi_2, input_dim, perm_op=perm_op)
+
+    data_test = [[[-7.04493841, 9.60285858],
+                  [-13.14389003, -13.21854157],
+                  [-3.21137961, -1.28593644]],
+                 [[10.40324933, -0.80540584],
+                  [16.54752459, 0.70355361],
+                  [6.410207, -10.63175183],
+                  [2.96613799, -11.97463568]],
+                 [[4.85041719, -2.93820024],
+                  [2.15379915, -5.39669696],
+                  [5.83968556, -5.67350982],
+                  [5.25955172, -6.36860269]]]
+
+    tf_data_test = tf.ragged.constant(data_test, ragged_rank=1)
+
+    RN.predict(tf_data_test)
+
+Once RN is properly trained (which we skip in this documentation) it can be used to make predictions.
```

Comment:
I've adapted Mathieu's original tutorial on the synthetic data so that it illustrates the use (including setup and training) of a RipsNet architecture. So, as @tlacombe  suggested, I think it may be nice to include and link to this tutorial somewhere.

Reply:
I've opened a PR to include the tutorial notebook into https://github.com/GUDHI/TDA-tutorial/ so that we can then link to it.